### ISSUE TITLE
Unexpected behavior with groupby on single-row dataframe?

### ISSUE THREAD

**Issue by dwyatte** [pydata/pandas#11741]
Do single-row dataframes (not series) get special treatment in some way? It seems you can do arbitrary groupby operations on them on non-existent columns without errors.

```
df1 = pd.DataFrame(np.random.randn(1,4), columns=list('ABCD'))
df2 = pd.DataFrame(np.random.randn(2,4), columns=list('ABCD'))

In [3]: df1.groupby('asdf')
Out[3]: pandas.core.groupby.DataFrameGroupBy object at 0x1139acb10

In [4]: df2.groupby('asdf')
KeyError: 'asdf'
```

**Comment by nbonnotte** [comment]
Yeah, there is a problem. I discovered that while working on issue #11640

I've a PR ready

**Comment by nbonnotte** [comment]
Solved with PR #12063

**Comment by dwyatte** [comment]
Great, thanks!
